ArmaforcesMissionBot/Handlers/ReactionHandler.cs

```diff
@@ -0,0 +1,333 @@
+                            if (newDescription.Count == 2)
+                                newEmbed.WithDescription(newDescription[0] + newDescription[1]);
+                            else if (newDescription.Count == 1)
+                                newEmbed.WithDescription(newDescription[0]);
+
+                            if (embed.Footer.HasValue)
+                                newEmbed.WithFooter(embed.Footer.Value.Text);
+                            else
+                                newEmbed.WithFooter(team.Pattern);
+
+                            await teamMsg.ModifyAsync(x => x.Embed = newEmbed.Build());
+                        }
+                        else
+                        {
+                            await teamMsg.RemoveReactionAsync(reaction.Emote, reaction.User.Value);
+                        }
+                    }
+                    else if (team.Slots.Any(x => x.Emoji == reaction.Emote.ToString()))
+                    {
+                        var teamMsg = await channel.GetMessageAsync(message.Id) as IUserMessage;
+                        await teamMsg.RemoveReactionAsync(reaction.Emote, reaction.User.Value);
+                    }
+                }
+            }
+            finally
+            {
+                mission.Access.Release();
+            }
+        }
+
+        private async Task AddedReactionToRolesChannel(Cacheable<IUserMessage, ulong> message, ISocketMessageChannel channel, SocketReaction reaction)
+        {
+            var fullMessage = await channel.GetMessageAsync(message.Id) as IUserMessage;
+
+            var messageDescription = fullMessage.Embeds.Single().Description;
+            ulong rank;
+            var reactionsMatches = MiscHelper.GetRankArrayMatchesFromText(messageDescription);
+            var rankForEmoji = reactionsMatches.Find(x => x.Item1 == reaction.Emote.ToString());
+
+            try
+            {
+                string regexPattern = @"\<\@\&([0-9]+)\>";
+                var rankMatches = Regex.Matches(rankForEmoji.Item2, regexPattern, RegexOptions.IgnoreCase | RegexOptions.RightToLeft).First().ToString();
+                rank = ulong.Parse(Regex.Matches(rankMatches, "[0-9]+", RegexOptions.IgnoreCase | RegexOptions.RightToLeft).First().Value);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"Parsing ranks failed.");
+                return;
+            }
+
+            try
+            {
+                var role = (channel as SocketGuildChannel).Guild.GetRole(rank);
+
+                await (reaction.User.Value as SocketGuildUser).AddRoleAsync(role);
+
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Adding rank failed.");
+            }
+
+            return;
+        }
+
+        private async Task RemovedReactionToSignUpsChannel(Cacheable<IUserMessage, ulong> message, ISocketMessageChannel channel, SocketReaction reaction, SignupsData signups)
```

Comment:
I think `RemovedReactionToSignUpsChannel` is a bad name pattern. Reactions are added to messages in channels, not to channels. I'd suggest renaming such methods to `RemovedReactionInSignupsChannel` (notice `Signups` vs `SignUps`).

Reply:
Done.